diff -up lightdm-1.0.1/Makefile.am.disable-tests lightdm-1.0.1/Makefile.am
--- lightdm-1.0.1/Makefile.am.disable-tests	2011-10-07 11:48:53.927051926 +0200
+++ lightdm-1.0.1/Makefile.am	2011-10-07 11:49:58.660602119 +0200
@@ -5,7 +5,7 @@ endif
 if COMPILE_LIBLIGHTDM_QT
 SUBDIRS += liblightdm-qt
 endif
-SUBDIRS += data greeters po src doc tests utils
+SUBDIRS += data greeters po src doc utils
 
 DISTCHECK_CONFIGURE_FLAGS = --enable-introspection --enable-gtk-doc
 
