diff -up lightdm-test/configure.fix-rpath lightdm-test/configure
--- lightdm-test/configure.fix-rpath	2011-10-07 21:45:25.680831300 +0200
+++ lightdm-test/configure	2011-10-07 22:43:14.860360766 +0200
@@ -10113,7 +10113,7 @@ aix[4-9]*)
   version_type=linux
   need_lib_prefix=no
   need_version=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   if test "$host_cpu" = ia64; then
     # AIX 5 supports IA64
     library_names_spec='${libname}${release}${shared_ext}$major ${libname}${release}${shared_ext}$versuffix $libname${shared_ext}'
@@ -10356,16 +10356,16 @@ freebsd* | dragonfly*)
     ;;
   freebsd3.[01]* | freebsdelf3.[01]*)
     shlibpath_overrides_runpath=yes
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     ;;
   freebsd3.[2-9]* | freebsdelf3.[2-9]* | \
   freebsd4.[0-5] | freebsdelf4.[0-5] | freebsd4.1.1 | freebsdelf4.1.1)
     shlibpath_overrides_runpath=no
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     ;;
   *) # from 4.6 on, and DragonFly
     shlibpath_overrides_runpath=yes
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     ;;
   esac
   ;;
@@ -10378,7 +10378,7 @@ gnu*)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 haiku*)
@@ -10391,7 +10391,7 @@ haiku*)
   shlibpath_var=LIBRARY_PATH
   shlibpath_overrides_runpath=yes
   sys_lib_dlsearch_path_spec='/boot/home/config/lib /boot/common/lib /boot/system/lib'
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 hpux9* | hpux10* | hpux11*)
@@ -10403,7 +10403,7 @@ hpux9* | hpux10* | hpux11*)
   case $host_cpu in
   ia64*)
     shrext_cmds='.so'
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     dynamic_linker="$host_os dld.so"
     shlibpath_var=LD_LIBRARY_PATH
     shlibpath_overrides_runpath=yes # Unless +noenvvar is specified.
@@ -10418,7 +10418,7 @@ hpux9* | hpux10* | hpux11*)
     ;;
   hppa*64*)
     shrext_cmds='.sl'
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     dynamic_linker="$host_os dld.sl"
     shlibpath_var=LD_LIBRARY_PATH # How should we handle SHLIB_PATH
     shlibpath_overrides_runpath=yes # Unless +noenvvar is specified.
@@ -10451,7 +10451,7 @@ interix[3-9]*)
   dynamic_linker='Interix 3.x ld.so.1 (PE, like ELF)'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 irix5* | irix6* | nonstopux*)
@@ -10488,7 +10488,7 @@ irix5* | irix6* | nonstopux*)
   shlibpath_overrides_runpath=no
   sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 # No shared lib support for Linux oldld, aout, or coff.
@@ -10544,7 +10544,7 @@ fi
   # This implies no fast_install, which is unacceptable.
   # Some rework will be needed to allow for fast_install
   # before this can be enabled.
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
 
   # Append ld.so.conf contents to the search path
   if test -f /etc/ld.so.conf; then
@@ -10569,7 +10569,7 @@ netbsdelf*-gnu)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   dynamic_linker='NetBSD ld.elf_so'
   ;;
 
@@ -10588,7 +10588,7 @@ netbsd*)
   fi
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 newsos6)
@@ -10606,7 +10606,7 @@ newsos6)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   dynamic_linker='ldqnx.so'
   ;;
 
@@ -10668,7 +10668,7 @@ solaris*)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   # ldd complains unless libraries are executable
   postinstall_cmds='chmod +x $lib'
   ;;
@@ -10725,7 +10725,7 @@ sysv5* | sco3.2v5* | sco5v6* | unixware*
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   if test "$with_gnu_ld" = yes; then
     sys_lib_search_path_spec='/usr/local/lib /usr/gnu/lib /usr/ccs/lib /usr/lib /lib'
   else
@@ -10747,7 +10747,7 @@ tpf*)
   library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 uts4*)
@@ -14504,7 +14504,7 @@ aix[4-9]*)
   version_type=linux
   need_lib_prefix=no
   need_version=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   if test "$host_cpu" = ia64; then
     # AIX 5 supports IA64
     library_names_spec='${libname}${release}${shared_ext}$major ${libname}${release}${shared_ext}$versuffix $libname${shared_ext}'
@@ -14745,16 +14745,16 @@ freebsd* | dragonfly*)
     ;;
   freebsd3.[01]* | freebsdelf3.[01]*)
     shlibpath_overrides_runpath=yes
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     ;;
   freebsd3.[2-9]* | freebsdelf3.[2-9]* | \
   freebsd4.[0-5] | freebsdelf4.[0-5] | freebsd4.1.1 | freebsdelf4.1.1)
     shlibpath_overrides_runpath=no
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     ;;
   *) # from 4.6 on, and DragonFly
     shlibpath_overrides_runpath=yes
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     ;;
   esac
   ;;
@@ -14767,7 +14767,7 @@ gnu*)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 haiku*)
@@ -14780,7 +14780,7 @@ haiku*)
   shlibpath_var=LIBRARY_PATH
   shlibpath_overrides_runpath=yes
   sys_lib_dlsearch_path_spec='/boot/home/config/lib /boot/common/lib /boot/system/lib'
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 hpux9* | hpux10* | hpux11*)
@@ -14792,7 +14792,7 @@ hpux9* | hpux10* | hpux11*)
   case $host_cpu in
   ia64*)
     shrext_cmds='.so'
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     dynamic_linker="$host_os dld.so"
     shlibpath_var=LD_LIBRARY_PATH
     shlibpath_overrides_runpath=yes # Unless +noenvvar is specified.
@@ -14807,7 +14807,7 @@ hpux9* | hpux10* | hpux11*)
     ;;
   hppa*64*)
     shrext_cmds='.sl'
-    hardcode_into_libs=yes
+    hardcode_into_libs=no
     dynamic_linker="$host_os dld.sl"
     shlibpath_var=LD_LIBRARY_PATH # How should we handle SHLIB_PATH
     shlibpath_overrides_runpath=yes # Unless +noenvvar is specified.
@@ -14840,7 +14840,7 @@ interix[3-9]*)
   dynamic_linker='Interix 3.x ld.so.1 (PE, like ELF)'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 irix5* | irix6* | nonstopux*)
@@ -14877,7 +14877,7 @@ irix5* | irix6* | nonstopux*)
   shlibpath_overrides_runpath=no
   sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
   sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 # No shared lib support for Linux oldld, aout, or coff.
@@ -14933,7 +14933,7 @@ fi
   # This implies no fast_install, which is unacceptable.
   # Some rework will be needed to allow for fast_install
   # before this can be enabled.
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
 
   # Append ld.so.conf contents to the search path
   if test -f /etc/ld.so.conf; then
@@ -14958,7 +14958,7 @@ netbsdelf*-gnu)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   dynamic_linker='NetBSD ld.elf_so'
   ;;
 
@@ -14977,7 +14977,7 @@ netbsd*)
   fi
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 newsos6)
@@ -14995,7 +14995,7 @@ newsos6)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   dynamic_linker='ldqnx.so'
   ;;
 
@@ -15057,7 +15057,7 @@ solaris*)
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   # ldd complains unless libraries are executable
   postinstall_cmds='chmod +x $lib'
   ;;
@@ -15114,7 +15114,7 @@ sysv5* | sco3.2v5* | sco5v6* | unixware*
   soname_spec='${libname}${release}${shared_ext}$major'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=yes
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   if test "$with_gnu_ld" = yes; then
     sys_lib_search_path_spec='/usr/local/lib /usr/gnu/lib /usr/ccs/lib /usr/lib /lib'
   else
@@ -15136,7 +15136,7 @@ tpf*)
   library_names_spec='${libname}${release}${shared_ext}$versuffix ${libname}${release}${shared_ext}$major $libname${shared_ext}'
   shlibpath_var=LD_LIBRARY_PATH
   shlibpath_overrides_runpath=no
-  hardcode_into_libs=yes
+  hardcode_into_libs=no
   ;;
 
 uts4*)
@@ -19045,7 +19045,7 @@ postinstall_cmds='`$ECHO "$postinstall_c
 postuninstall_cmds='`$ECHO "$postuninstall_cmds" | $SED "$delay_single_quote_subst"`'
 finish_cmds='`$ECHO "$finish_cmds" | $SED "$delay_single_quote_subst"`'
 finish_eval='`$ECHO "$finish_eval" | $SED "$delay_single_quote_subst"`'
-hardcode_into_libs='`$ECHO "$hardcode_into_libs" | $SED "$delay_single_quote_subst"`'
+hardcode_into_libs=no
 sys_lib_search_path_spec='`$ECHO "$sys_lib_search_path_spec" | $SED "$delay_single_quote_subst"`'
 sys_lib_dlsearch_path_spec='`$ECHO "$sys_lib_dlsearch_path_spec" | $SED "$delay_single_quote_subst"`'
 hardcode_action='`$ECHO "$hardcode_action" | $SED "$delay_single_quote_subst"`'
@@ -20292,7 +20292,7 @@ finish_cmds=$lt_finish_cmds
 finish_eval=$lt_finish_eval
 
 # Whether we should hardcode library paths into libraries.
-hardcode_into_libs=$hardcode_into_libs
+hardcode_into_libs=no
 
 # Compile-time system search path for libraries.
 sys_lib_search_path_spec=$lt_sys_lib_search_path_spec
