diff -uNdr lightdm-0.9.2.original/tests/src/Makefile.am lightdm-0.9.2/tests/src/Makefile.am
--- lightdm-0.9.2.original/tests/src/Makefile.am	2011-07-25 06:04:29.000000000 +0400
+++ lightdm-0.9.2/tests/src/Makefile.am	2011-07-31 03:59:09.825263441 +0400
@@ -49,7 +49,7 @@
 	$(GLIB_LIBS)
 
 test-qt-greeter_moc.cpp: test-qt-greeter.h
-	moc $< -o $@
+	moc-qt4 $< -o $@
 test_qt_greeter_SOURCES = test-qt-greeter.cpp test-qt-greeter.h test-qt-greeter_moc.cpp status.c status.h
 test_qt_greeter_CXXFLAGS = \
 	-I$(top_srcdir)/liblightdm-qt \
@@ -57,7 +57,8 @@
 test_qt_greeter_LDADD = \
 	-L$(top_builddir)/liblightdm-qt \
 	-llightdm-qt-1 \
-	$(LIGHTDM_QT_GREETER_LIBS)
+	$(LIGHTDM_QT_GREETER_LIBS) \
+	$(XCB_LIBS)
 
 test_session_SOURCES = test-session.c status.c status.h
 test_session_CFLAGS = \

diff -uNdr lightdm-0.9.2.original/tests/src/Makefile.in lightdm-0.9.2/tests/src/Makefile.in
--- lightdm-0.9.2.original/tests/src/Makefile.in	2011-07-26 13:00:17.000000000 +0400
+++ lightdm-0.9.2/tests/src/Makefile.in	2011-07-31 03:59:36.322472937 +0400
@@ -88,7 +88,8 @@
 	test_qt_greeter-test-qt-greeter.$(OBJEXT) \
 	test_qt_greeter-test-qt-greeter_moc.$(OBJEXT) status.$(OBJEXT)
 test_qt_greeter_OBJECTS = $(am_test_qt_greeter_OBJECTS)
-test_qt_greeter_DEPENDENCIES = $(am__DEPENDENCIES_1)
+test_qt_greeter_DEPENDENCIES = $(am__DEPENDENCIES_1) \
+	$(am__DEPENDENCIES_1)
 test_qt_greeter_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CXX \
 	$(AM_LIBTOOLFLAGS) $(LIBTOOLFLAGS) --mode=link $(CXXLD) \
 	$(test_qt_greeter_CXXFLAGS) $(CXXFLAGS) $(AM_LDFLAGS) \
@@ -397,7 +398,8 @@
 test_qt_greeter_LDADD = \
 	-L$(top_builddir)/liblightdm-qt \
 	-llightdm-qt-1 \
-	$(LIGHTDM_QT_GREETER_LIBS)
+	$(LIGHTDM_QT_GREETER_LIBS) \
+	$(XCB_LIBS)
 
 test_session_SOURCES = test-session.c status.c status.h
 test_session_CFLAGS = \
@@ -968,7 +970,7 @@
 
 
 test-qt-greeter_moc.cpp: test-qt-greeter.h
-	moc $< -o $@
+	moc-qt4 $< -o $@
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.

diff -uNdr lightdm-0.9.2.original/greeters/qt/Makefile.am lightdm-0.9.2/greeters/qt/Makefile.am
--- lightdm-0.9.2.original/greeters/qt/Makefile.am	2011-07-22 10:50:53.000000000 +0400
+++ lightdm-0.9.2/greeters/qt/Makefile.am	2011-07-31 03:52:04.421837204 +0400
@@ -4,16 +4,16 @@
 dist_xgreeters_DATA = lightdm-qt-greeter.desktop
 
 greeter_moc.cpp: greeter.h
-	moc $< -o $@
+	moc-qt4 $< -o $@
 loginprompt_moc.cpp: loginprompt.h
-	moc $< -o $@
+	moc-qt4 $< -o $@
 panel_moc.cpp: panel.h
-	moc $< -o $@
+	moc-qt4 $< -o $@
 
 ui_loginprompt.h: loginprompt.ui
-	uic $< -o $@
+	uic-qt4 $< -o $@
 ui_panel.h: panel.ui
-	uic $< -o $@
+	uic-qt4 $< -o $@
 
 loginprompt.cpp: ui_loginprompt.h
 panel.cpp: ui_panel.h

diff -uNdr lightdm-0.9.2.original/liblightdm-qt/Makefile.am lightdm-0.9.2/liblightdm-qt/Makefile.am
--- lightdm-0.9.2.original/liblightdm-qt/Makefile.am	2011-07-19 11:25:36.000000000 +0400
+++ lightdm-0.9.2/liblightdm-qt/Makefile.am	2011-07-31 03:52:04.423837220 +0400
@@ -10,11 +10,11 @@
 	user_moc.cpp 
 
 greeter_moc.cpp: QLightDM/Greeter
-	moc $< -o $@
+	moc-qt4 $< -o $@
 session_moc.cpp: QLightDM/Session
-	moc $< -o $@
+	moc-qt4 $< -o $@
 user_moc.cpp: QLightDM/User
-	moc $< -o $@
+	moc-qt4 $< -o $@
 
 liblightdm_qt_1_la_LIBADD = $(LIBLIGHTDM_QT_LIBS)
 liblightdm_qt_1_la_CXXFLAGS = $(LIBLIGHTDM_QT_CFLAGS) \
